### PR TITLE
Fix secret copy

### PR DESCRIPTION
**Pull request by teresaromero**
# Proposed changes

Secrets declared at the manifest, or via the cmd flag `--secret` where removed when creating a Copy of the BuildInfo struct.
- This PR adds Secrets and other fields from BuildInfo that where being removed as nil pointers.
- Added integration and unit test for Copy function.

On the other hand, this PR also implements the use of templates at secret files, this way a secret can be mounted not only having  a local file, but using a template for the secret and having envs that will be replaced when running the build and mounting the secret.
- Parses the content of the secret file into builtkit as a temp file

### CODE REVIEW

File: pkg/model/dev.go
Comment (jLopezbarb) [inline pkg/model/dev.go:1301]: I think we will always forget to add the new fields here... Maybe we should use something like https://github.com/jinzhu/copier to perform this task. The problem with these modules is that they use reflect inside which can cause runtime errors.

File: pkg/model/dev.go
Comment (pchico83) [inline pkg/model/dev.go:1321]: Instead of passing the original content of the file, let's create a temporal file (and remove it with a defer after the build) with the content of the file after applying env substitutiton.
For example, for `.npmrc`, you would have this file as a secret:
```
egistry=https://registry.npmjs.org/
//***.jfrog.io/artifactory/api/npm/:_authToken=${NPM_TOKEN}
@scope**:registry=https://***.jfrog.io/artifactory/api/npm/private-npm
```
you won't need to commit your token, okteto will replace the token using okteto secrets or your local env vars

File: pkg/model/dev.go
Comment (teresaromero) [reply pkg/model/dev.go]: i've used marshall unmarshall method, is this an option? seems to work fine

Comment (AgustinRamiroDiaz) [reply pkg/model/dev.go]: @teresaromero we could also have a silent error with marshalling because it needs the fields to be exported. If some day we add an unexported field, it won't be copied

Comment (teresaromero) [reply pkg/model/dev.go:1301]: i've reverted to loop between fields and pointers.
@jLopezbarb it is a risk to forget... but this struct is not changing that much, and eventually the bug will appear when we use differnt build options if we add some new.... I prefer this than other silent errors or adding a new library....

Comment (jLopezbarb) [reply pkg/model/dev.go:1301]: I'm ok with it, I just wanted to make sure that this was going to be a problem when implementing new fields. Right now it only affects buildInfo so it's a thing we can deal with it but if we ever have several structs to copy we should take a look for one of the solutions

**Comment by ifbyol** [comment]
@teresaromero is there a ticket linked to this? Is this a fast track or a normal issue?

**Comment by teresaromero** [comment]
> @teresaromero is there a ticket linked to this? Is this a fast track or a normal issue?

This was opened after @pchico83  reported at slack to unblock the feature for secrets

**Review by ifbyol — APPROVED** [review]

File: pkg/cmd/build/build.go
Comment (teresaromero) [reply pkg/cmd/build/build.go:130]: 🤔  i've taken a look and i've moved out the `parseTempSecrets` the folder creation, this way the defer will clear all in case of error. WDYT?

**Review by jLopezbarb — APPROVED** [review]

File: pkg/cmd/build/build.go
Comment (ifbyol) [reply pkg/cmd/build/build.go:130]: yeah, makes sense!

**Review by ifbyol — APPROVED** [review]

**Pull request merged**
